src/_pytest/logging.py

```diff
@@ -596,10 +596,10 @@ def pytest_collection(self):
             if self.log_cli_handler:
                 self.log_cli_handler.set_when("collection")
 
-            if self.log_file_handler is not None:
```

Comment:
This `if` should stay as is -- what should is the `else`, to this:

```py
else:
    # Add a dummy handler to ensure logging.root.handlers is not empty.
    # If it were empty, then a `logging.warning()` call (and similar) during collection
    # would trigger a `logging.basicConfig()` call, which would add a `StreamHandler`
    # handler, which would cause all subsequent logs which reach the root to be also
    # printed to stdout, which we don't want (issue #6240).
    with catching_logs(logging.NullHandler()):
        yield
```

Reply:
looks good, done